src/main/java/edu/cornell/kfs/concur/service/impl/ConcurAccountValidationServiceImpl.java

```diff
@@ -40,28 +39,46 @@ public ValidationResult validateConcurAccountInfo(ConcurAccountInfo concurAccoun
                 concurAccountInfo.getSubAccountNumber(),
                 concurAccountInfo.getObjectCode(),
                 concurAccountInfo.getSubObjectCode(),
-                concurAccountInfo.getProjectCode());
+                concurAccountInfo.getProjectCode(),
+                true);
     }
+    
+    @Override
+    public ValidationResult validateConcurAccountInfoObjectCodeNotRequired(ConcurAccountInfo concurAccountInfo) {
+        return checkAccountingString(
+                concurAccountInfo.getChart(),
+                concurAccountInfo.getAccountNumber(),
+                concurAccountInfo.getSubAccountNumber(),
+                concurAccountInfo.getObjectCode(),
+                concurAccountInfo.getSubObjectCode(),
+                concurAccountInfo.getProjectCode(),
+                false);
+    }
+
 
-    public ValidationResult checkAccountingString(String chartOfAccountsCode, String accountNumber, String subAccountNumber, String objectCode, String subObjectCode, String projectCode) {
-        ValidationResult requiredFieldsValidationResult = checkRequiredAccountInfo(chartOfAccountsCode, accountNumber, objectCode);
+    public ValidationResult checkAccountingString(String chartOfAccountsCode, String accountNumber, String subAccountNumber, String objectCode, String subObjectCode, String projectCode, boolean objectCodeRequired) {
```

Comment:
If you added the ConcurBatchUtilityService method suggested from my earlier comment, you might be able to refactor this class to avoid these boolean method arguments.